src/api-service/__app__/onefuzzlib/orm.py

```diff
@@ -223,6 +222,33 @@ def raw(
         return result
 
 
+B = TypeVar("B", bound=BaseModel)
+
+
+def hide_secrets(data: B, hider: Callable[[SecretData], SecretData]) -> B:
+    for field in data.__fields__:
+        field_data = getattr(data, field)
+
+        if isinstance(field_data, SecretData):
+            field_data = hider(field_data)
+        elif isinstance(field_data, list):
+            if len(field_data) > 0:
+                if not isinstance(field_data[0], BaseModel):
```

Comment:
Is this meant to check that `field_data : [BaseModel]`, or something else?

If the former, I'm not sure why checking the first element is enough (i.e. why we can assume that `field_data` is a homogenous list, if it is a list).

Reply:
Ah, is it because we have typed `BaseModel` fields, and we assume our field-typing regime enforces homogenous lists?

If so, that seems better, but still not quite enough (couldn't they be `Union[]` values?). Again, I might just be missing something.

Reply:
Oh, you just _moved_ this code. I guess it'd still be good to confirm 😄 .

Reply:
The assumption was if the first field was a BaseModel, all the fields would be.  However, I agree that's a naïve approach.  

Addressed.